populus/project.py

```diff
@@ -159,46 +162,60 @@ def user_config(self):
 
     @property
     def project_config(self):
-
         if self._project_config_cache is None:
             project_config = Config(
                 config=self._project_config,
-                schema=self._project_config_schema,
             )
             project_config.unref()
             self._project_config_cache = project_config
 
+            # schema validation
+            # partial project config must be merged to get the entire schema
+            if self._user_config_cache is None:
+                self.user_config
```

Comment:
I think this kind of cache population produces difficult to read and understand code because it *magically* populates a cache....